lyft-helper.js

```diff
@@ -1,26 +1,60 @@
-/* TODO:
--
--
--
+var lyftMethods = require('lyftPrivateMethods');
+var baseURL = 'https://api.lyft.com/v1/'; // on which path is added.
 
-/
+var lyftPhoneAuth = function(phoneNumberString) {
+  var url = baseURL + lyftMethods.phoneAuth.path;
+  var headers = lyftMethods.phoneAuth.headers; // or headers() ?
+  var body = lyftMethods.phoneAuth.body(phoneNumberString);
 
-*/
+  fetch(url, {
+    method: 'POST',
+    headers: headers,
+    body: body // JSON.stringify ? or done in privateMethods ?
+  }).then( function(res) {
+    return res.json();
+  }).then( function(data) {
+    console.log('successful phoneNumber post LYFT', data);
+    // TODO: next step. this also depends on the user input of 2FA code.
+    // response irrelevant unless we pass through session
+  }).catch( function(err) {
+    console.log('error post of phoneNumber LYFT', err);
+  });
+};
 
-// function won't be used here - instead on webview.
-// possible refactor to ES6 with React needed.
+var lyftPhoneCodeAuth = function(fourDigitCode, session) {
+  var url = baseURL + lyftMethods.phoneCodeAuth.path;
+  var headers = lyftMethods.phoneCodeAuth.headers(session);
+  var body = lyftMethods.phoneCodeAuth.body(fourDigitCode);
 
-// endpoints, headers, body parameters.
-var lyftMethods = require('lyftPrivateMethods');
+  fetch(url, {
+    method: 'POST',
+    headers: headers,
+    body: body // JSON.stringify ? or done in privateMethods ?
+  }).then( function(res) {
+    return res.json();
+  }).then( function(data) {
+    console.log('successful phoneCodeAuth post LYFT', data);
 
-var lyftPhoneAuth = function(phoneNumberString) {
-  // user gives us his phone number, we fetch(url, {method: POST});
-  // response irrelevant
-  // lyftMethods.phoneAuthHeaders, lyftMethods.phoneAuthBody (function)
+    // TODO: next step.
+
+  }).catch( function(err) {
+    console.log('error post of phoneCodeAuth LYFT', err);
+  });
 };
 
-var lyftPhoneCodeAuth = function(fourDigitCode) {
-  // user inputs 4 digit code, we fetch(url, {method: POST});
-  // we take response and store in lyftPrivateMethods ?
-  // lyftMethods.phoneCodeAuthHeaders, lyftMethods.phoneCodeAuthBody (function)
+var getCost = function(token, session, endAddress, endLat, endLng, startLat, startLng, startAddress) {
```

Comment:
may want to collapse those address, lat, and long variables into one object that gets passed